src/ipfsd-daemon.js

```diff
@@ -270,36 +270,40 @@ class Daemon {
   /**
    * Kill the `ipfs daemon` process.
    *
-   * First `SIGTERM` is sent, after 10.5 seconds `SIGKILL` is sent
-   * if the process hasn't exited yet.
+   * If the HTTP API is established, then send 'shutdown' command; otherwise,
+   * process.kill(`SIGTERM`) is used.  In either case, if the process
+   * does not exit after 10.5 seconds then a `SIGKILL` is used.
    *
    * @param {function()} callback - Called when the process was killed.
    * @returns {undefined}
    */
   killProcess (callback) {
     // need a local var for the closure, as we clear the var.
+    const self = this
```

Comment:
Self is not needed.

Reply:
Disagree, it is needed in `subprocess.once('exit', ...`

`this` should not be used in a lambdaexpression!  See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions